rasa/cli/utils.py

```diff
@@ -124,6 +129,68 @@ def create_output_path(
         return os.path.join(output_path, file_name)
 
 
+def button_to_string(button, idx=0):
+    """Create a string representation of a button."""
+
+    title = button.pop("title", "")
+
+    if "payload" in button:
+        payload = " ({})".format(button.pop("payload"))
+    else:
+        payload = ""
+
+    # if there are any additional attributes, we append them to the output
+    if button:
+        details = " - {}".format(json.dumps(button, sort_keys=True))
+    else:
+        details = ""
+
+    button_string = "{idx}: {title}{payload}{details}".format(
+        idx=idx + 1, title=title, payload=payload, details=details
+    )
+
+    return button_string
+
+
+def element_to_string(element, idx=0):
+    """Create a string representation of an element."""
+    title = element.pop("title", "")
+
+    element_string = "{idx}: {title} - {element}".format(
+        idx=idx + 1, title=title, element=json.dumps(element, sort_keys=True)
+    )
+
+    return element_string
+
+
+def button_choices_from_message_data(
+    message: Dict[Text, Any], allow_free_text_input: bool = True
+) -> Question:
+    """Return list of choices to present to the user.
+
+    If allow_free_text_input is True, an additional option is added
+    at the end along with the template buttons that allows the user
+    to type in free text.
+    """
+    choices = [
+        button_to_string(button, idx)
+        for idx, button in enumerate(message.get("buttons"))
+    ]
+    if allow_free_text_input:
+        choices.append(FREE_TEXT_INPUT_PROMPT)
+    return choices
+
+
+def payload_from_button_question(button_question: Question) -> Text:
+    """Prompt user with a button question and returns the nlu payload."""
+    response = button_question.ask()
+    if response == FREE_TEXT_INPUT_PROMPT:
+        # Free text input option is chosen
+        return response
```

Comment:
I've played around with it and know it works, but i'm confused as to why this works, If `response ==  FREE_TEXT_INPUT_PROMPT` doesn't this just return `FREE_TEXT_INPUT_PROMPT`?

Reply:
ah wait I understand now. Maybe the comment should say something like `# return FREE_TEXT_INPUT_PROMPT to prompt for free text input question` or somehting like that

Reply:
I've reorganized the function a bit differently now.